src/assets/token.ts

```diff
@@ -0,0 +1,159 @@
+			amount += BigInt(unlockingObj.amount);
+		}
+	}
+
+	if (amount > 0) {
+		return [{ module: MODULE_NAME_DPOS, amount: String(amount) }];
+	}
+	return [];
+};
+
+export const createLegacyReserveAccount = async (
+	accounts: Account[],
+	legacyAccounts: LegacyStoreEntry[],
+	tokenID: string,
+): Promise<UserStoreEntry> => {
+	const legacyReserveAccount: Account | undefined = accounts.find(account =>
+		ADDRESS_LEGACY_RESERVE.equals(account.address),
+	);
+
+	let legacyReserveAmount = legacyReserveAccount ? legacyReserveAccount.token.balance : AMOUNT_ZERO;
+
+	for (const account of legacyAccounts) {
+		legacyReserveAmount += BigInt(account.balance);
+	}
+	const lockedBalances = legacyReserveAccount ? await getLockedBalances(legacyReserveAccount) : [];
+	lockedBalances.push({
+		module: MODULE_NAME_LEGACY,
+		amount: String(legacyReserveAmount),
+	});
+	const legacyReserve = {
+		address: ADDRESS_LEGACY_RESERVE_HEX,
+		tokenID,
+		availableBalance: String(
+			legacyReserveAccount ? legacyReserveAccount.token.balance : AMOUNT_ZERO,
+		),
+		lockedBalances,
+	};
+
+	return legacyReserve;
+};
+
+export const createUserSubstoreArray = async (
+	accounts: Account[],
+	legacyAccounts: LegacyStoreEntry[],
+	tokenID: string,
+): Promise<UserStoreEntry[]> => {
+	const userSubstore: UserStoreEntry[] = [];
+	for (const account of accounts) {
+		if (!ADDRESS_LEGACY_RESERVE.equals(account.address)) {
+			const userObj = {
+				address: account.address.toString('hex'),
+				tokenID,
+				availableBalance: String(account.token.balance),
+				lockedBalances: await getLockedBalances(account),
+			};
+			userSubstore.push(userObj);
+		}
+	}
+
+	const legacyReserveAccount = await createLegacyReserveAccount(accounts, legacyAccounts, tokenID);
+	userSubstore.push(legacyReserveAccount);
+
+	const sortedUserSubstore = userSubstore.sort((a: UserStoreEntry, b: UserStoreEntry) => {
+		const bufferObjA = {
+			address: Buffer.from(a.address, 'hex'),
+			tokenID: Buffer.from(a.tokenID, 'hex'),
+		};
+		const bufferObjB = {
+			address: Buffer.from(b.address, 'hex'),
+			tokenID: Buffer.from(b.tokenID, 'hex'),
+		};
+
+		return bufferObjA.address.equals(bufferObjB.address)
+			? bufferObjA.tokenID.compare(bufferObjB.tokenID)
+			: bufferObjA.address.compare(bufferObjB.address);
+	});
+
+	return sortedUserSubstore.map(entry => ({
+		...entry,
+		address: getLisk32AddressFromAddress(Buffer.from(entry.address, 'hex')),
+	}));
+};
+
+export const createSupplySubstoreArray = async (
+	accounts: Account[],
+	tokenID: string,
+): Promise<SupplyStoreEntry[]> => {
+	let totalLSKSupply = AMOUNT_ZERO;
+	for (const account of accounts) {
+		totalLSKSupply += BigInt(account.token.balance);
+		const lockedBalances = await getLockedBalances(account);
+		totalLSKSupply = lockedBalances.reduce(
+			(accumulator, lockedBalance) => accumulator + BigInt(lockedBalance.amount),
+			totalLSKSupply,
+		);
+	}
+	const LSKSupply = { tokenID, totalSupply: String(totalLSKSupply) };
```

Comment:
Here the `legacyReserveAmount` is missing.

Reply:
This change is handled with the PR https://github.com/LiskHQ/lisk-migrator/pull/67